OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/DbxFileActions.java

```diff
@@ -131,16 +137,30 @@ public void openFromDropbox(DropboxRevisionsTopComponent drtc, RevisionDisplayTo
     /**
      * @return List with user's files or empty List if error occurs
      */
-    public List<DbxEntryDto> getDbxTexEntries(DbxClient client) {
+    public List<DbxEntryDto> getDbxTexEntries(DbxClientV2 client) {
         List<DbxEntryDto> dbxEntries = new ArrayList<>();
 
         if (client == null) {
             return dbxEntries;
         }
 
         try {
-            for (DbxEntry entry : client.searchFileAndFolderNames("/", TEX_EXTENSION)) {
-                dbxEntries.add(new DbxEntryDto(entry));
+            ListFolderResult result = client.files().listFolderBuilder("").withRecursive(true).start();
+            while (true) {
+                for (Metadata metadata : result.getEntries()) {
+                    if (metadata instanceof FileMetadata) {
+                        String name = metadata.getName();
+                        if (name.endsWith(TEX_EXTENSION)) {
+                            dbxEntries.add(new DbxEntryDto((FileMetadata) metadata));
+                        }
+                    }
+                }
+
+                if (!result.getHasMore()) {
```

Comment:
Good idea

Reply:
Hmm...
In one way or another, it seems that I cannot run from the `while (true)` because the `result.getHasMore()` check needs to come before the `client.files().listFolderContinue(result.getCursor())` call...

Notice that the starting point:
```
ListFolderResult result = client.files().listFolderBuilder("").withRecursive(true).start();
```
is different from the continuation:
```
result = client.files().listFolderContinue(result.getCursor());
```